From cae20f6aedce28223c9a2bd108168a87ce4c32ad Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:23 -0400
Subject: [fs] btrfs: make DEV_INFO ioctl available to anyone

Message-id: <1407179109-13714-317-git-send-email-sandeen@redhat.com>
Patchwork-id: 87318
O-Subject: [RHEL7.1 PATCH 316/422] btrfs: make DEV_INFO ioctl available to anyone
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 7d6213c5a764d0591afeb9faa3f7ee18e8e74c25
Author: David Sterba <dsterba@suse.cz>
Date:   Thu Apr 24 15:09:10 2014 +0200

    btrfs: make DEV_INFO ioctl available to anyone

    This ioctl provides basic info about the devices that can be obtained in
    other ways (eg. sysfs), there's no reason to restrict it to
    CAP_SYSADMIN.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index c6c8e35..92a5184 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -2593,9 +2593,6 @@ static long btrfs_ioctl_dev_info(struct btrfs_root *root, void __user *arg)
  int ret = 0;
  char *s_uuid = NULL;
 
- if (!capable(CAP_SYS_ADMIN))
-  return -EPERM;
-
  di_args = memdup_user(arg, sizeof(*di_args));
  if (IS_ERR(di_args))
   return PTR_ERR(di_args);
-- 
1.7.1